### DISCUSSION TITLE
Pods not being recreated after Secret change

### DISCUSSION THREAD

**Discussion by marcorichetta** [fluxcd/flux2#2489] [Q&A] [answered]
Hi!

I have fluxcd already configured (BTW, it's great).
I modified a couple of secrets in a `secret.yaml` file, created the `sealed-secret.yaml` and pushed that to the master branch.
FluxCD detected it but only modified the secret on the cluster, **not redeploying the pod attached to that secret.**

Notifications received on a Slack channel:

> **gitrepository/dev.flux-system**
> Fetched revision: master/06e35cffaf0e71ced077e898bbac898830102eea
> 
> **kustomization/dev.flux-system**
> SealedSecret/namespace-name/deployment-name configured
> **revision**
> master/06e35cffaf0e71ced077e898bbac898830102eea

- Is this the expected behaviour?
- Could this be solved by changing the reconciliation strategy? https://github.com/fluxcd/flux2/discussions/2324#discussioncomment-2109244

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
> Is this the expected behaviour?

Yes, if you want deployments to change you can use SOPS with kustomize secrets generator. https://fluxcd.io/docs/components/kustomize/kustomization/#kustomize-secretgenerator

**Reply by marcorichetta** [reply]
May I ask 
- What's the difference between having the config in the kustomization.yaml vs on a separated file like secret.yaml?
- Is it possible to recreate deployments that use a key defined in the changed secret?

**Reply by kingdonb** [reply]
Yes @marcorichetta – the pattern is the same as in https://fluxcd.io/docs/guides/helmreleases/#refer-to-values-in-configmaps-generated-with-kustomize

In the example provided, the "secret" is a ConfigMap and the "deployment" is a HelmRelease, but the principle is the same. If you use a configmap generator, the configmap will have a generated hash suffix appended to the name. If you use a KustomizeConfig like the one provided, then you can update any references to the Secret in the Deployment with the generated name.

These are in-built features of Kustomize.
